### PR TITLE
Add recipe for scine-database

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by awvwgk** [comment]
@conda-forge/help-python-c This recipes is ready for review (C++ library + pybind11 extension module). Package is split to keep build times and package sizes manageable.

For the testing I have to create a mongod process since the tests can't do this themselves for some reason, which adds some complexity and output clutter.

- scine-database check: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=532050&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=5030
- scine-database-python check: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=532050&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=6197

File: recipes/scine-database-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  imports:
+    - scine_database
+  source_files:
+    - src/Database/Python/Tests/
+  requires:
+    - pip
+    - pytest
+  commands:
+    - pip check
+    - $RECIPE_DIR/test.sh
```
Comment (carterbox) [inline recipes/scine-database-python/meta.yaml:60]:
```suggestion
```
Rename the test to `run_test.sh` and put pip check inside the script.

File: recipes/scine-database-python/test.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -ex
+
```
Comment (carterbox) [inline recipes/scine-database-python/test.sh:4]:
```suggestion

pip check

```

File: recipes/scine-database/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    - eigen
+    - gtest
+    - gmock
+    # Needed for eigen detection
+    - libblas
+    - libmongocxx
+    - mongodb
+    - scine-utilsos >=5.0.0
+  run:
+    - libmongocxx
```
Comment (carterbox) [inline recipes/scine-database/meta.yaml:43]:
```suggestion
```
libmongocxx has a run_export

**Review by carterbox — CHANGES_REQUESTED** [review]
Looks very good, just some minor suggestions.

**Comment by awvwgk** [comment]
@conda-forge/help-c-cpp Requested changes are addressed

**Review by carterbox — APPROVED** [review]

**Pull request merged**